include/rocksdb/utilities/backupable_db.h

```diff
@@ -256,24 +275,36 @@ class BackupEngine {
   static Status Open(Env* db_env, const BackupableDBOptions& options,
                      BackupEngine** backup_engine_ptr);
 
-  // same as CreateNewBackup, but stores extra application metadata
-  // Flush will always trigger if 2PC is enabled.
-  // If write-ahead logs are disabled, set flush_before_backup=true to
-  // avoid losing unflushed key/value pairs from the memtable.
+  // same as CreateNewBackup, but stores extra application metadata.
   virtual Status CreateNewBackupWithMetadata(
-      DB* db, const std::string& app_metadata, bool flush_before_backup = false,
-      std::function<void()> progress_callback = []() {}) = 0;
+      DB* db, const std::string& app_metadata,
+      const CreateBackupOptions& options = CreateBackupOptions()) = 0;
+
+  // keep here for backward compatibility.
+  virtual Status CreateNewBackupWithMetadata(
+      DB* db, const std::string& app_metadata, bool flush_before_backup,
+      std::function<void()> progress_callback = []() {}) {
+    CreateBackupOptions options;
+    options.flush_before_backup = flush_before_backup;
+    options.progress_callback = progress_callback;
+    return CreateNewBackupWithMetadata(db, app_metadata, options);
+  }
 
   // Captures the state of the database in the latest backup
   // NOT a thread safe call
-  // Flush will always trigger if 2PC is enabled.
-  // If write-ahead logs are disabled, set flush_before_backup=true to
-  // avoid losing unflushed key/value pairs from the memtable.
-  virtual Status CreateNewBackup(DB* db, bool flush_before_backup = false,
+  virtual Status CreateNewBackup(
+      DB* db, const CreateBackupOptions& options = CreateBackupOptions()) {
```

Comment:
Can we skip the default and force users to provide an option, like all other options? Also, We tend to put options before other arguments, so it is probably to be better to switch between db and options.

Reply:
I followed https://github.com/facebook/rocksdb/pull/6602/files#diff-d6f7c7da4019d974b2c3c68b4ff2ba84R116 to be consistent

Reply:
Can we change...

Reply:
I'll change both

Reply:
Please put changing the other function in a separate PR if you want.  I think there is no urgency to do that though.